### PR TITLE
Improve API for some Snapshot methods

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Both methods make the `Snapshot` instance unuseable and so should take ownership of it to limit potential misuse from the users.

### CODE REVIEW

File: gtk4/src/snapshot.rs [hunk 1]
```diff
@@ -27,4 +27,19 @@ impl Snapshot {
     pub fn push_debug(&self, message: &str) {
         unsafe { ffi::gtk_snapshot_push_debug(self.to_glib_none().0, message.to_glib_none().0) }
     }
+
+    #[doc(alias = "gtk_snapshot_to_node")]
+    pub fn to_node(self) -> Option<gsk::RenderNode> {
```
Comment (sdroege) [inline gtk4/src/snapshot.rs:32]: Should this really be `Option`s or `Result<T, glib::BoolError>`?

Comment (GuillaumeGomez) [reply gtk4/src/snapshot.rs:32]: What the `BoolError` should be? It's not necessarily an error when it returns `NULL`.

Comment (sdroege) [reply gtk4/src/snapshot.rs:32]: Ah it's not? Then it should indeed be an `Option`

**Comment by GuillaumeGomez** [comment]
Anything else to be done in here?

**Comment by bilelmoussaoui** [comment]
> Anything else to be done in here?

I am waiting to see if the ToGlibPtrMove traits will land soon-ish or not, we can update the PR before landing it then.

**Comment by sdroege** [comment]
It's there now but the gir-related changes are still pending. @GuillaumeGomez what's the status on that?

**Comment by GuillaumeGomez** [comment]
Still need to do it.

**Comment by GuillaumeGomez** [comment]
I finally took the time to update it. I needed to update the gir submodules too to get the change so the first

**Comment by bilelmoussaoui** [comment]
Closing in favour of https://github.com/gtk-rs/gtk4-rs/pull/1097

**Pull request closed without merging**